Base management of POIs

Joxit opened the pull request:
# POIs classification

## Shop `#F3C600` (text `#C17900`)

| class   | subclass            |
| ------- | ------------------- |
| car     |                     |
| shop    |                     |
| bicycle |                     |
|         | mobile_phone        |
|         | florist             |
|         | convenience         |
|         | clothes             |
|         | shoes               |
|         | boutique            |
|         | furniture           |
|         | hardware            |
|         | jewelry             |
|         | optician            |
|         | interior_decoration |
|         | watches             |
|         | mall                |
|         | marketplace         |
|         | video_games         |
|         | toys                |
|         | alcohol_shop        |
|         | grocery             |
|         | hairdresser         |
|         | gift                |

## Food & Drink `#EF8000`

| class | subclass      |
| ----- | ------------- |
|       | fast_food     |
|       | confectionary |
|       | ice_cream     |
|       | beverages     |
|       | bbq           |
|       | restaurant    |
|       | bar           |
|       | bakery        |
|       | cafe          |
|       | beer          |

## Transport `#8665E7`

| class     | subclass               |
| --------- | ---------------------- |
| aerialway | station                |
| railway   | station                |
| railway   | tram_stop              |
|           | halt                   |
|           | subway_entrance        |
|           | train_station_entrance |
|           | bus_stop               |
|           | subway                 |
|           | ferry_terminal         |
|           | entrance               |
|           | airport                |

## Service `#495063`

| class            | subclass         |
| ---------------- | ---------------- |
| cemetery         |                  |
| college          |                  |
| information      |                  |
| library          |                  |
| post             |                  |
| school           |                  |
| town_hall        |                  |
| harbor           |                  |
| laundry          |                  |
| lodging          |                  |
| place_of_worship |                  |
| place_of_worship | christian        |
| place_of_worship | muslim           |
| place_of_worship | jewish           |
|                  | bank             |
|                  | car_rental       |
|                  | car_repair       |
|                  | police           |
|                  | embassy          |
|                  | fire_station     |
|                  | public_building  |
|                  | prison           |
|                  | place_of_worship |
|                  | bicycle_rental   |

## Entertainment `#E93865`

| class      | subclass    |
| ---------- | ----------- |
| attraction |             |
|            | theatre     |
|            | arts_centre |
|            | cinema      |
|            | nightclub   |
|            | museum      |
|            | gallery     |
|            | zoo         |
|            | theme_park  |
|            | aquarium    |

## Nature `#76BC54` (text `#538C37`)

| class    | subclass    |
| -------- | ----------- |
| campsite |             |
|          | park        |
|          | dog_park    |
|          | garden      |
|          | basin       |
|          | reservoir   |
|          | picnic_site |

## Health `#4AA0E7`

| class    | subclass   |
| -------- | ---------- |
| hospital |            |
|          | pharmacy   |
|          | veterinary |
|          | doctors    |
|          | clinic     |
|          | dentist    |

## Unclassified `#5C6F84`

| class    | subclass      |
| -------- | ------------- |
| barrier  |               |
| castle   |               |
| golf     |               |
| swimming |               |
| stadium  |               |
|          | toilets       |
|          | telephone     |
|          | waste_basket  |
|          | equestrian    |
|          | swimming_pool |
|          | food_court    |
|          | playground    |
|          | artwork       |
|          | viewpoint     |
|          | sports_centre |
|          | fuel          |
|          | recycling     |
|          | gate          |
|          | cycle_barrier |
|          | stile         |
|          | shelter       |
|          | sally_port    |
|          | monument      |
|          | athletics     |
|          | pitch         |
|          | stadium       |

# First Comment

* Add polygon management: 
    * universities (same as schoool)
![univertisty](https://user-images.githubusercontent.com/5153882/47558598-cbb45800-d913-11e8-962e-b75de41b9535.png)
    * aerodrome (gray 50%)
![aerodrome](https://user-images.githubusercontent.com/5153882/47558607-cf47df00-d913-11e8-9fb7-439605097048.png)

* Color harmonization between polygon and icons for school
![school](https://user-images.githubusercontent.com/5153882/47558613-d40c9300-d913-11e8-8329-7a12700b9e37.png)

* Add label color for fire_station (same as police)
![fire-station](https://user-images.githubusercontent.com/5153882/47558625-df5fbe80-d913-11e8-8cda-6487e85e01fa.png)

* Fallback icon based on class for: alcohol_shop, attraction, barrier, beer, bicycle, campsite, car, castle, cemetery, college, golf, grocery, harbor, hospital, information, library, laundry, lodging, post, school, shop, stadium, swimming, town_hall
![shop](https://user-images.githubusercontent.com/5153882/47558634-e4247280-d913-11e8-8285-769f790d6389.png)
![information](https://user-images.githubusercontent.com/5153882/47558641-e5ee3600-d913-11e8-9b12-b7896b565eb6.png)

In order to use fallback icons, you must put specific icons based on subclasses before the fallback icon based on class.
All fallback incons are at the end of the file.

Joxit committed:
* Add polygon management for universities (same as schoool), aerodrome (gray 50%)
* Color harmonization between polygon and icons for school
* Add label color for fire_station (same as police)
* Fallback icon based on class for: alcohol_shop, attraction, barrier, beer, bicycle, campsite, car, castle, cemetery, college, golf, grocery, harbor, hospital, information, library, laundry, lodging, post, school, shop, stadium, swimming, town_hall

nlehuby commented:
Thanks for these changes :+1:

The grey creates contrast issues with the buildings, on both the airports and the schools (but maybe it can be fixed by tuning the opacity)
![screenshot_2018-10-26 debug your style 2](https://user-images.githubusercontent.com/919962/47569732-a552e480-d934-11e8-9ba3-a682f4bb4322.png)
![screenshot_2018-10-26 debug your style](https://user-images.githubusercontent.com/919962/47569746-ad128900-d934-11e8-9c1f-3bee29bea26e.png)


The cemetaries and the theme parks are still to do:
![screenshot_2018-10-26 debug your style 1](https://user-images.githubusercontent.com/919962/47569847-037fc780-d935-11e8-9f33-e38e8f195cf0.png)

Joxit committed:
Change landuse-aerodrome color for a bluish background

Joxit commented:
Oups, you're right, I changed airports for a bluish background (to be consistent with the icon)

![capture d ecran de 2018-10-29 18-28-23](https://user-images.githubusercontent.com/5153882/47668448-c459ac80-dba8-11e8-9ba0-a60644f850d8.png)

I think schools are ok no?

Joxit committed:
Update Transport & Health icons

- change background colors for aerodrome & health
- change icon colors for transports & health
- add background for bus_station

Joxit commented:
Update Transport & Health icons

- change background colors for aerodrome & health
![stage-2 1-aerodrome](https://user-images.githubusercontent.com/5153882/47799525-bcc21100-dd2a-11e8-8596-c41f215ccd2e.png)
![stage-2 1-hospital](https://user-images.githubusercontent.com/5153882/47799527-bcc21100-dd2a-11e8-8826-4d5e2ca464b0.png)
- change icon colors for transports & health
![stage-2 1-mixed](https://user-images.githubusercontent.com/5153882/47799528-bcc21100-dd2a-11e8-826c-8c7c405823d3.png)
- add background & icon for bus_station
![stage-2 1-bus_station](https://user-images.githubusercontent.com/5153882/47799803-4c67bf80-dd2b-11e8-8ad5-56111e7ac012.png)

More about classification [here](#issue-226037007)

nlehuby commented:
so far so good. The `#4AA0E7` for health is a bit strange at first look but not that bad actually.

Joxit committed:
Update `Shop` & `Food & Drink`

- Change icon colors for `Shop` & `Food & Drink`
- Fix florist icon (wrong size/stroke/align)
- Create correct icon for commercial and gift

Joxit commented:
Update `Shop` & `Food & Drink`

- Change icon colors for `Shop` & `Food & Drink` (note hairdresser as shop, is it ok ?) 
![stage-2 2-food](https://user-images.githubusercontent.com/5153882/48008109-4dc52d80-e119-11e8-82f2-48cba64f0556.png)
![stage-2 2-hairdresser](https://user-images.githubusercontent.com/5153882/48008161-62a1c100-e119-11e8-97ea-554473e6df60.png)
- Fix florist icon (wrong size/stroke/align)
![stage-2 2-florist](https://user-images.githubusercontent.com/5153882/48008063-3dad4e00-e119-11e8-8111-bc9402a8ac06.png)
- Create correct icon for commercial and gift
![stage-2 2-commercial](https://user-images.githubusercontent.com/5153882/48008071-4140d500-e119-11e8-9f49-874d7bbd081d.png)
![stage-2 2-gift](https://user-images.githubusercontent.com/5153882/48008093-4736b600-e119-11e8-968a-7f87c21326b6.png)

More about classification [here](#issue-226037007)

Joxit committed:
Update `Service` & `Entertainment`

- Change icon colors for `Service` & `Entertainment`
- Create correct icon for building

Joxit commented:
Update `Service` & `Entertainment`
- Change icon colors for `Service` & `Entertainment`
    - Cemetery
![stage-2 3-cemetery](https://user-images.githubusercontent.com/5153882/48081542-9996d600-e1f0-11e8-9590-41044ec4f052.png)
    - Eiffel Tower
![stage-2 3-eiffel-tower](https://user-images.githubusercontent.com/5153882/48081543-9996d600-e1f0-11e8-98c8-b5b72edf3c78.png)
    - Opera
![stage-2 3-opera](https://user-images.githubusercontent.com/5153882/48081544-9996d600-e1f0-11e8-8e92-d277ce0f406d.png)
    - School & community center
![stage-2 3-school center](https://user-images.githubusercontent.com/5153882/48081547-9a2f6c80-e1f0-11e8-860e-a9d3603fbdd1.png)
- Create correct icon for building
![stage-2 3-public-building](https://user-images.githubusercontent.com/5153882/48081545-9996d600-e1f0-11e8-97b0-e718d81ada75.png)

Joxit committed:
Update `Nature` & `Unclassified`

Change icon colors for `Nature` & `Unclassified`
- Create food-court icon (picnic-site with different color)
- Create correct icon for barrier, shelter, dam, ranger-station, pitch, 
recycling

Joxit commented:
Here are the last categories `Nature` & `Unclassified`

- Change icon colors for `Nature` & `Unclassified`
![stage-2 4-park](https://user-images.githubusercontent.com/5153882/48147693-340e1c80-e2b8-11e8-99fc-9acb2512895a.png)
![stage-2 4-roland-garros](https://user-images.githubusercontent.com/5153882/48147694-340e1c80-e2b8-11e8-9202-fea1c6140733.png)
![stage-2 4-sport](https://user-images.githubusercontent.com/5153882/48147695-34a6b300-e2b8-11e8-8142-f6c0cc391485.png)
- Create food-court icon (picnic-site with different color)
- Create correct icon for barrier, shelter, dam, ranger-station, pitch, recycling
![stage-2 4-barrier](https://user-images.githubusercontent.com/5153882/48147677-2bb5e180-e2b8-11e8-9cb5-58ac05ab5d68.png)

Joxit committed:
Fix toilets subclass, icon is toilet-11

nlehuby commented on icons.yml:
```diff
@@ -7,390 +7,327 @@ mappings:
-    color: '#8A9CF6'
+    color: '#495063'
   - subclass: artwork
     iconName: attraction-11
```
artwork are almost street furniture, and they are really too ugly in pink : can we have them in Unclassified colour instead of Entertainment one ?
![screenshot_2018-11-08 luxembourg](https://user-images.githubusercontent.com/919962/48212210-69c80980-e37b-11e8-970e-274d7ec5b62e.png)

Joxit replied:
Should I change artwork only or artwork + attraction ? 

Example with only artwork
![artwork only](https://user-images.githubusercontent.com/5153882/48268236-b9b6d700-e434-11e8-951e-374c01df6a5e.png)

nlehuby commented on icons.yml:
```diff
@@ -7,390 +7,327 @@ mappings:
+    iconName: pitch-11
+    color: '#5C6F84'
+  - subclass: pitch
+    iconName: pitch-11
```
the `pitch` picto looks like an emergency exit one, can we switch to the `soccer` one (while we enhance the tile schema to better handle sports item) ?

nlehuby commented on style.json:
```diff
@@ -439,6 +442,44 @@
+        "fill-color": "#8665E7",
+        "fill-opacity": 0.06
+      }
+    },
```
there are still a few POI polygons missing: theme_park and zoo (low opacity pink ?), stadium pitch and playground (low opacity Unclassified colour ?)

nlehuby reviewed (commented):
And the default picto also need changes: it should be in "Unclassified" colour and should not be a star.

Joxit committed:
Update shop text color and add some background

- Add landuse-entertainment and landse-pitch background
- Update landse-commercial background
- Update shop text color
- Update default marker

Joxit commented:
Update shop text color and add some background

- Add landuse-entertainment and landse-pitch background
![pitch in hospital](https://user-images.githubusercontent.com/5153882/48276679-e96fda00-e448-11e8-960e-4bbc5185586f.png)
![pitch](https://user-images.githubusercontent.com/5153882/48276680-e96fda00-e448-11e8-8f54-850536c70d3c.png)
![attraction](https://user-images.githubusercontent.com/5153882/48276682-ea087080-e448-11e8-81aa-5862207f658e.png)
- Update landuse-commercial background (same as industrial)
![landuse-commercial](https://user-images.githubusercontent.com/5153882/48276731-05737b80-e449-11e8-88c7-1f03beea5051.png)
- Update shop text color
![shop](https://user-images.githubusercontent.com/5153882/48276760-17edb500-e449-11e8-9268-5cfd0bd6be06.png)
- Update default marker
![default marker](https://user-images.githubusercontent.com/5153882/48276822-423f7280-e449-11e8-893b-5889ff7200ab.png)

Joxit committed:
Icons from #51 recategorized

- Car icon move from service to shop.
- New icons :
  - `car-rental` and `car-repair` are in service
  - `confectionary` is in food & drink
  - `convenience-store`, `furniture`, `hardware`, `jewlery-store`,
`optician`, `paint` and `watch` are in shop
  - `viewpoint` is in unclassified
- Fix : add default icon for art_gallery class

Joxit commented:
Icons from #51 recategorized

- Car icon move from service to shop.
![car](https://user-images.githubusercontent.com/5153882/48341807-57491b00-e66e-11e8-8c62-bfd633e7ea7a.png)
- New icons :
  - `car-rental` and `car-repair` are in service
![car-repair](https://user-images.githubusercontent.com/5153882/48341771-40a2c400-e66e-11e8-9f80-d8843983f8ea.png)
  - `confectionary` is in food & drink
  - `convenience-store`, `furniture`, `hardware`, `jewlery-store`, `optician`, `paint` and `watch` are in shop
![image](https://user-images.githubusercontent.com/5153882/48341911-a1320100-e66e-11e8-9c75-53f0bfde988f.png)
![image](https://user-images.githubusercontent.com/5153882/48342284-88761b00-e66f-11e8-88e4-5eabc233858a.png)
  - `viewpoint` is in unclassified
![image](https://user-images.githubusercontent.com/5153882/48342311-9b88eb00-e66f-11e8-9550-e8c102937d43.png)

Joxit committed:
Update pitch background and park color labels

Joxit commented:
- New pitch background
![image](https://user-images.githubusercontent.com/5153882/48410698-ab700000-e73f-11e8-9953-a8c478419a10.png)
- New park text color
![image](https://user-images.githubusercontent.com/5153882/48410750-d8bcae00-e73f-11e8-90fd-012ecfc9ad5e.png)
